tests/phpunit/tests/date/currentTime.php

```diff
@@ -157,4 +180,36 @@ public function test_should_work_with_deprecated_timezone() {
 		$this->assertSame( gmdate( $format ), $current_time_gmt, 'The dates should be equal [3]' );
 		$this->assertSame( $datetime->format( $format ), $current_time, 'The dates should be equal [4]' );
 	}
+
+	/**
+	 * @ticket 57998
+	 *
+	 * @dataProvider data_partial_hour_timezones_with_timestamp
+	 *
+	 * @param float $partial_hour Partial hour GMT offset to test.
+	 */
+	public function test_partial_hour_timezones_with_timestamp( $partial_hour ) {
+		update_option( 'gmt_offset', $partial_hour );
+
+		$expected = time() + (int) ( $partial_hour * HOUR_IN_SECONDS );
+
+		// phpcs:ignore WordPress.DateTime.CurrentTimeTimestamp.Requested
+		$this->assertSame( $expected, current_time( 'timestamp' ) );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_partial_hour_timezones_with_timestamp() {
+		return array(
+			'+13:45' => array( 13.45 ), // New Zealand, Chatham Islands.
+			'+9:30'  => array( 9.5 ), // Australian Northern Territory.
+			'+05:30' => array( 5.5 ), // India and Sri Lanka.
+			'+05:45' => array( 5.45 ), // Nepal.
+			'-03:30' => array( -3.30 ), // Canada, Newfoundland.
+			'-09:30' => array( -9.30 ), // French Polynesia, Marquesas Islands.
+		);
```

Comment:
The Northern Territory and India and Sri Lanka are using `.5` to indicate half an hour but the others are all using `.30` for the half hour and `.45` for the hour.

Is this intentional due to how PHP is representing the data?